### DISCUSSION TITLE
How to set drag limit for objects relative to another object

### DISCUSSION THREAD

**Discussion by xorb** [fabricjs/fabric.js#8607] [Q&A]
Here is my current approach, but it is not smooth.

```ts
  const updatePosition = () => {
    image.setCoords()
    cropper.setCoords();
    const isIn = cropper.isContainedWithinObject(image);
    if(!isIn){
    	if(image._stateProperties){
      	image.left = image._stateProperties.left;
        image.top = image._stateProperties.top;
      }
    } else {
      image.setCoords();
      image.saveState();
    }
    
  }
  image.on("moving", updatePosition)
```

Here is demo link: https://jsfiddle.net/xorbmoon/3ruaL9gs/114/

**Comment by ShaMan123** [comment]
Your solution is too simple.
That is why the image gets "stuck".
You need to handle more cases. check against all corners. do min/max calculations.
Don't use state! It is deprecated.
BTW we will be publishing cropping controls in the near future.

**Reply by xorb** [reply]
Is it coming on v6?

**Comment by iocoker** [comment]
@ShaMan123 Will the planned cropping be similar to this: https://konvajs.org/docs/sandbox/Scale_Image_To_Fit.html (with scaling) or just a regular crop to size?
I've been struggling with the crop/scaling concept for a while

**Reply by ShaMan123** [reply]
not to offend konva, but I saw how it works (@asturur gave me a peek), can't be compared.
You can control everything with ease, all common cropping UX.
